packages/internal/bridge/sdk/src/tokenBridge.ts

```diff
@@ -169,41 +181,79 @@ export class TokenBridge {
     let sourceChainFee: ethers.BigNumber = ethers.BigNumber.from(0);
     let approvalFee: ethers.BigNumber = ethers.BigNumber.from(0);
     let bridgeFee: ethers.BigNumber = ethers.BigNumber.from(0);
-    const imtblFee: ethers.BigNumber = ethers.BigNumber.from(0);
+
+    let direction: BridgeDirection;
+    if (req.action === BridgeFeeActions.DEPOSIT) {
+      direction = {
+        sourceChainId: req.sourceChainId,
+        destinationChainId: req.destinationChainId,
+        action: BridgeFeeActions.DEPOSIT,
+      };
+    } else if (req.action === BridgeFeeActions.WITHDRAW) {
+      direction = {
+        sourceChainId: req.sourceChainId,
+        destinationChainId: req.destinationChainId,
+        action: BridgeFeeActions.WITHDRAW,
+      };
+    } else {
+      // Throws if you call this function when action is FINALISE_WITHDRAW
+      throw new BridgeError('Invalid action', BridgeErrorType.INTERNAL_ERROR);
+    }
+
+    if (
+      !(isValidDeposit(direction, this.config.bridgeInstance) || isValidWithdraw(direction, this.config.bridgeInstance))
+    ) {
+      throw new BridgeError('Invalid direction', BridgeErrorType.INTERNAL_ERROR);
+    }
 
     // Get approval fee
     if ('token' in req && req.token.toUpperCase() !== NATIVE) {
-      if (req.sourceChainId === this.config.bridgeInstance.rootChainID) {
+      if (isValidDeposit(direction, this.config.bridgeInstance)) {
         approvalFee = calculateGasFee(feeData, BridgeMethodsGasLimit.APPROVE_TOKEN);
-      } else if (req.sourceChainId === this.config.bridgeInstance.childChainID
-        && isWrappedIMX(req.token, this.config.bridgeInstance.childChainID)) {
+      } else if (isWithdrawWrappedIMX(req.token, direction, this.config.bridgeInstance)) {
         // On child chain, only WIMX requires approval.
         approvalFee = calculateGasFee(feeData, BridgeMethodsGasLimit.APPROVE_TOKEN);
       }
     }
 
     // Get source fee & bridge fee
+    let axelarGasLimit;
+    if (req.action === BridgeFeeActions.DEPOSIT) {
+      sourceChainFee = calculateGasFee(feeData, BridgeMethodsGasLimit.DEPOSIT_SOURCE);
+      axelarGasLimit = BridgeMethodsGasLimit.DEPOSIT_DESTINATION;
+    } else {
+      sourceChainFee = calculateGasFee(feeData, BridgeMethodsGasLimit.WITHDRAW_SOURCE);
+      axelarGasLimit = BridgeMethodsGasLimit.WITHDRAW_DESTINATION;
```

Comment:
I would have assumed we'd be using dynamic gas estimates on these?

Reply:
Looks like `getDepositOrWithdrawFee` is used by `getFeePrivate` (the old interface), where we do the static estimation. Fees obtained with `getFeePrivate` are used for illustration purposes only so does not involve transaction building.